### PR TITLE
[IMP] tests: use concat demuxer to encode frames

### PR DESCRIPTION
**Pull request by d-fence**
When the screencast argument is used to produce a video file of failing
tests, the framerate is computed with a KISS average.
This results in a misleading video flow. For example, if a tour step is
stuck, the average could show a smooth transition instead of the reality.

With this commit, the real duration of frames are used to produce the
video. The result is a more realistic video flow.

### CODE REVIEW

File: odoo/tests/common.py [hunk 1]
```diff
@@ -886,8 +886,19 @@ def _save_screencast(self, prefix='failed'):
...
+            nb_frames = len(self.screencast_frames)
+            concat_script_path = os.path.join(self.screencasts_dir, fname.replace('.mp4', '.txt'))
+            with open(concat_script_path, 'w') as concat_file:
+                for i in range(nb_frames):
+                    frame_file_path = os.path.join(self.screencasts_frames_dir, self.screencast_frames[i]['file_path'])
+                    concat_file.write("file '%s'\n" % frame_file_path)
+                    if i < nb_frames - 1:
+                        duration = self.screencast_frames[i+1]['timestamp'] - self.screencast_frames[i]['timestamp']
+                    else:
+                        duration = time.time() - self.screencast_frames[i]['timestamp']
```
Comment (Xavier-Do) [inline odoo/tests/common.py:898]:
```suggestion
                    end = time.time() if i == nb_frames - 1 else self.screencast_frames[i+1]['timestamp']
                    duration = end - self.screencast_frames[i]['timestamp']

```

File: odoo/tests/common.py [hunk 2]
```diff
@@ -886,8 +886,19 @@ def _save_screencast(self, prefix='failed'):
...
+                    concat_file.write('duration %s\n' % duration)
```
Comment (Xavier-Do) [inline odoo/tests/common.py:899]: maybe something like 
```python3
concat_file.write("file '%s'\nduration %s\n" % (frame_file_path, duration))
```

File: odoo/tests/common.py [hunk 3]
```diff
@@ -899,7 +910,8 @@ def start_screencast(self):
             os.makedirs(self.screencasts_dir, exist_ok=True)
             self.screencasts_frames_dir = os.path.join(self.screencasts_dir, 'frames')
             os.makedirs(self.screencasts_frames_dir, exist_ok=True)
-        self._websocket_send('Page.startScreencast', params={'maxWidth': 1024, 'maxHeight': 576})
+        width, height = (int(size) for size in self.window_size.split('x'))
+        self._websocket_send('Page.startScreencast', params={'maxWidth': width, 'maxHeight': height})
```
Comment (Xavier-Do) [inline odoo/tests/common.py:914]: I'm not sure params are needed anymore, I guess that the default max is the window size. 
```suggestion
        self._websocket_send('Page.startScreencast')
```

**Review by Xavier-Do — CHANGES_REQUESTED** [review]

File: odoo/tests/common.py (hunk 3 above)
Comment (d-fence) [reply odoo/tests/common.py:914]: You cannot be sure with the [experimental devtools protocol ](https://chromedevtools.github.io/devtools-protocol/tot/Page/#method-startScreencast).
But let's try

**Comment by Xavier-Do** [comment]
robodoo r+

**Pull request closed without merging**